app/server/lib/OIDCConfig.ts

```diff
@@ -135,28 +188,33 @@ export class OIDCConfig {
   }
 
   public async handleCallback(sessions: Sessions, req: express.Request, res: express.Response): Promise<void> {
-    const mreq = req as RequestWithLogin;
+    let mreq;
+    try {
+      mreq = this._getRequestWithSession(req);
+    } catch(err) {
+      log.warn("OIDCConfig callback:", err.message);
+      return this._sendErrorPage(req, res);
+    }
+
     try {
       const params = this._client.callbackParams(req);
-      const { state, targetUrl } = mreq.session?.oidc ?? {};
-      if (!state) {
-        throw new Error('Login or logout failed to complete');
-      }
+      const { targetUrl } = mreq.session.oidc ?? {};
 
-      const codeVerifier = await this._retrieveCodeVerifierFromSession(req);
+      const checks = this._protectionManager.getCallbackChecks(mreq.session.oidc ?? {});
```

Comment:
We use `{}` if `oidc` isn't defined here.

Would it be better to check earlier if `oidc` is defined, and return if it isn't?

My interpretation is it's invalid for `oidc` to be undefined at this point, so we'd be fine to return an error page to the user? 

And then we can use `oidc` here without worrying about it.